Python testlib

jburel opened the pull request:
# What this PR does

Clean up ``testlib`` for first release.
Reduce the number of "import" methods so we do not overload the testing library
This will now be a "public" tool useful to test external app e.g. mapr

# Testing this PR

 * Check if further cleaning is required
 * Check that the tests are still green


# Related reading

See https://trello.com/c/ECAKLbUr/224-omero-testlib

jburel committed:
Reduce the number of "import_image" methods

jburel committed:
Adjust the tests accordingly

jburel committed:
flake8

sbesson commented:
As the semantics of all import methods is covered by the central `import_mif`. this looks fine from my perspective. A couple of additional comments in the perspective of giving this a public exposition:

- `import_mif` do we want to deprecate and rename the  public name to be simply `import_images` or even `import`?
- in the same vein, is there an argument for renaming `series_count` as `image_count` since series only refers to the internal implementation using BF fakes
- what's the use case of `seriesCount=0`? should we change the default to `1` instead as this would simplify most of the default client calls?
- in terms of best practice for the usage of the library, as per the method definition, we might want to pass arguments as key/value pairs rather than using implicit positional arguments

jburel commented:
@sbesson:
 * I did not yet check if there is a use case for ``seriescount = 0``, for me 1 should be the default
 * probably no need to deprecate since 5.3.0 will be the first public release of it so I will vote for import_images too

jburel committed:
Rename methods

jburel commented:
Actually ``seriesCount = 0`` seems to be used for plate

jburel committed:
Rename variable

jburel committed:
pep8 fix

sbesson commented:
That makes complete sense as `import_plates` overrides the classical `images/series` semantics of the `FakeReader` as the number of images is imposed by the SPW structure. If this is the major use case, I would definitely vote for upping the default to 1, reduce the amount of hardcoding across tests and specifically override in the case of plates. I will also check how the `FakeReader` handles the mismatch.

Update: after double checking the SPW semantics seems to unconditionally override `seriesCount` so passing a zero value is technically not even required.

jburel committed:
Rename pix to create_pixels

jburel committed:
rename variable

jburel committed:
Rename method and adjust the tests

jburel committed:
flake8

jburel committed:
Remove print

jburel commented:
I think we also need to remove the usage of  ``tinyTest.d3d.dv`` in the library before we release

jburel committed:
Replace usage of import_image w/o name

jburel committed:
Fix tests

jburel committed:
Add method to check the pixels type

jburel committed:
Move assert methods

jburel committed:
Remove usage of import_image method

jburel committed:
Improve tests

jburel committed:
Move method to import tinydv in web

test_marshal is the only place using tinydv.
Usage to tinydv should be removed at some point

jburel committed:
Adjust test and fix variable name

jburel committed:
Deprecate share test method

jburel committed:
Move method to test

jburel committed:
Move method to test_ishare

So we do not have to deprecate the method when we release
for the first time

jburel committed:
Review pixels type method

joshmoore commented on components/tools/OmeroPy/src/omero/testlib/__init__.py:
```diff
@@ -693,15 +693,13 @@ def create_pixels(self, x=10, y=10, z=10, c=3, t=50, pixels_type="int8",
             return 4
         if pixels_type in ["double"]:
             return 8
-        return -1
```
Perhaps this should throw? I assume that what would have happened if `-1` got passed to the server.

jburel replied:
it returns ``1`` by default. I pushed a commit to that effect afterwards so it did not change the existing expectation

atarkowska commented on components/tools/OmeroPy/src/omero/testlib/__init__.py:
```diff
@@ -1440,10 +1304,95 @@ def fullImport(self, client):
         assert 1 == len(rsp.pixels)
         return rsp
+
+    def assertWrite(self, mrepo2, filename, ofile):
```
as we already started to follow pep standards why these doesn't follow naming nomenclature?

jburel replied:
Will do that. Still few things to clean up.
Need to stabilise first the tests :-)

atarkowska commented on components/tools/OmeroPy/test/integration/test_ishare.py:
```diff
@@ -23,6 +23,25 @@
 class TestIShare(ITest):
 
+    @classmethod
+    def create_share(cls, description="", timeout=None,
```
if share is deprecated now, how about if we add warning to test helpers as well?

jburel replied:
good idea

atarkowska reviewed (commented):
in general this looks fine to me. What is the best strategy to rebase this library against m52 branch? I had difficulty to do it smoothly.

jburel committed:
Rename methods

jburel committed:
Fix test accordingly

jburel committed:
Deprecate method in test

jburel committed:
Raise Exception if type not supported

Fix method setting the type

sbesson commented on components/tools/OmeroPy/src/omero/testlib/__init__.py:
```diff
@@ -296,49 +298,17 @@ def import_image(self, filename=None, client=None, extra_args=None,
+    def import_fake_file(self, images_count=1, name=None, client=None,
+                         with_companion=False, skip="all", **kwargs):
+        """
+        Creates a fake file with a seriesCount of images, imports
```
`s/seriesCount/images_count`. Otherwise, the name of the new method makes sense and is representative of the called behavior.

jburel committed:
Fix doc

will-moore commented:
Not part of this PR, but this docstring needs a cleanup (still mentions Dataset and project):
```
    def make_file_annotation(self, name=None, binary=None, format=None,
                             client=None, ns=None):
        """
        Creates a new DatasetI instance and returns the persisted object.
        If no name has been provided, a UUID string shall be used.
        :param name: the name of the project
        :param client: The client to use to create the object
        :param ns: The namespace for the annotation
        """
```

jburel committed:
Review make_file_annotation method

jburel commented:
@will-moore: fix the description of the method and also adjust the code

jburel committed:
Add try/finally

jburel committed:
Fix test

jburel commented:
https://ci.openmicroscopy.org/job/OMERO-DEV-merge-integration-Python27/lastCompletedBuild/testReport/ no test failure due to this PR

will-moore commented:
Good to merge from me.
https://ci.openmicroscopy.org/job/OMERO-DEV-merge-integration-Python27/469/testReport/OmeroWeb.test.integration.test_tree/TestTree/test_marshal_images_thumb_version_thumb0_/ failures 
```AttributeError: 'TestTree' object has no attribute 'createTestImage' ``` need to be fixed in a different PR (probably https://github.com/openmicroscopy/openmicroscopy/pull/5138?) cc @aleksandra-tarkowska

jburel committed:
Merge remote-tracking branch 'origin/develop' into python-test

jburel committed:
Use new methods

will-moore commented:
Last commits look good.
All green today at https://ci.openmicroscopy.org/job/OMERO-DEV-merge-integration-Python27/471/
